File: requery-test/src/main/java/io/requery/test/immutables/Phone.java
package io.requery.test.immutables;

import io.requery.Entity;
import io.requery.ForeignKey;
import io.requery.Generated;
import io.requery.Key;
import org.immutables.value.Value;

@Value.Immutable
@Entity(model = "immutable")
public abstract class Phone {

    public static ImmutablePhone.Builder builder() {
        return ImmutablePhone.builder();
    }

    @Key @Generated
    public abstract int getId();
    public abstract String getPhoneNumber();
    public abstract boolean isNormalized();

    @ForeignKey(references = Person.class)
    public abstract int getOwnerId();

    // this method should not be processed
    public boolean isValid() {
        return getPhoneNumber() != null && isNormalized();
    }
}

File: requery-android/example-kotlin/src/main/kotlin/io/requery/android/example/app/model/Phone.kt
package io.requery.android.example.app.model

import android.os.Parcelable
import io.requery.*

@Entity
interface Phone : Parcelable, Persistable {
    @get:Key
    @get:Generated
    val id: Int

    var phoneNumber: String

    @get:ManyToOne
    var owner: Person
}
